HumanDynamicsEstimationLibrary/include/InverseVelocityKinematics.hpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef INVERSEVELOCITYKINEMATICS_HPP
+#define INVERSEVELOCITYKINEMATICS_HPP
+
+#include <Eigen/Dense>
+#include <Eigen/QR>
+#include <Eigen/SparseCholesky>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/Model/Model.h>
+
+#include <map>
+#include <memory>
+#include <yarp/os/LogStream.h>
+
+namespace hde {
+    class InverseVelocityKinematics;
+} // namespace hde
+
+class hde::InverseVelocityKinematics
+{
+protected:
+    class impl;
+    std::unique_ptr<impl> pImpl;
+
+public:
+    enum class InverseVelocityKinematicsResolutionMode
```

Comment:
Nit: this is already scoped with the class name, either move it outside the class, or rename to `ResolutionMode`.